tests/executors/test_executor.py

```diff
@@ -55,9 +55,11 @@ def sort_by(item):
                 (_, prio, _, _) = val
                 # Sort by priority (DESC), then date,task, try
                 return -prio, date, dag_id, task_id, try_number
-            sorted_queue = sorted(self.queued_tasks.items(), key=sort_by)
 
-            for (key, (_, _, _, simple_ti)) in sorted_queue:
+            open_slots = self.parallelism - len(self.running)
+            sorted_queue = sorted(self.queued_tasks.items(), key=sort_by)
+            for _ in range(min((open_slots, len(sorted_queue)))):
+                (key, (_, _, _, simple_ti)) = sorted_queue.pop()
```

Comment:
This change is just to  make the test executor behave more like the other ones that have limited parallelism right?

Reply:
Yes. We can't test overflowing if the executor "runs" all tasks so we had to create a bottleneck :).